### ISSUE TITLE
attempt to call field 'format' (a nil value)

### ISSUE THREAD

**Issue by AmeerTaweel** [wbthomason/packer.nvim#501]
- `nvim --version`: NVIM v0.6.0-dev+65-g46009499a
- `git --version`: 2.30.2
- Operating system/version: Ubuntu 21.04
- Terminal name/version: alacritty 0.9.0-dev

### Steps to reproduce

I don't know what happened, but whenever I open NeoVim, I get this error:

```
E5113: Error while calling lua chunk: ...im/site/pack/packer/start/packer.nvim/lua/packe
r/log.lua:61: attempt to call field 'format' (a nil value)
```

### Actual behaviour

Whenever I open NeoVim, I get this error:

```
E5113: Error while calling lua chunk: ...im/site/pack/packer/start/packer.nvim/lua/packe
r/log.lua:61: attempt to call field 'format' (a nil value)
```

### Expected behaviour

It should open normally. It used to do yesterday. I don't know why it's doing this.

### packer files

<details>
<summary>Plugin specification file(s)</summary>

Post or link your plugin specification files here, if you aren't able to provide a minimal
reproducer

</details>
```lua
require "globals"

local packerInstallPath = fn.stdpath("data").."/site/pack/packer/start/packer.nvim"
local packerRepoURL = "https://github.com/wbthomason/packer.nvim"

if fn.empty(fn.glob(packerInstallPath)) == 1 then
	fn.system({"git", "clone", packerRepoURL, packerInstallPath})
	exec.command "packadd packer.nvim"
end

local packer = require "packer"
local packerUtils = require "packer.util"

-- Define packages
local packerPackages = function(use)

	-- Packer can manage itself
	use "wbthomason/packer.nvim"

	-- [[ Themes ]]

	-- OneHalf
	use { "sonph/onehalf", rtp = "vim" }

	-- Gruvbox
	use "gruvbox-community/gruvbox"

	-- Dracula
	use { "dracula/vim", as = "dracula" }

	-- Nord
	use "arcticicestudio/nord-vim"

	-- [[ Git ]]

	-- Fugitive: Git for Vim
	use "tpope/vim-fugitive"

	use {
		"nvim-treesitter/nvim-treesitter",
		run = ":TSUpdate",
		config = function()
			require "plugin-config.nvim-treesitter"
		end
	}

	use {
		"p00f/nvim-ts-rainbow",
		wants = "nvim-treesitter",
		config = function()
			require "plugin-config.nvim-ts-rainbow"
		end
	}

    -- use "neovim/nvim-lspconfig"

    -- use {
    --     "kabouzeid/nvim-lspinstall",
    --     wants = "nvim-lspconfig",
    --     config = function()
    --         require "plugin-config.lsp.nvim-lspconfig"
    --     end
    -- }

	use {
		"hrsh7th/nvim-compe",
		config = function()
			require "plugin-config.nvim-compe"
		end
	}

	-- [[ Others ]]

    -- VIM-TMUX Navigator: Better VIM and TMUX navigation
    -- Should also installed in TMUX with TMUX Plugin Manager (TPM)
	use {
        "christoomey/vim-tmux-navigator",
		config = function()
			require "plugin-config.vim-tmux-navigator"
		end
	}

    -- Vim Eunuch: Unix Shell commands in Vim
    use "tpope/vim-eunuch"

    -- Window Swap: Swap split windows with ease
    use {
        "wesQ3/vim-windowswap",
		config = function()
			require "plugin-config.window-swap"
		end
    }
end

-- Define configurations
local packerGlobalConfig = {
	display = {
		open_fn = packerUtils.float
	}
}

-- Pass packages and configuration to Packer
packer.startup({ packerPackages, config = packerGlobalConfig })

-- Auto install missing packages automatically
exec.command "PackerInstall"
```
<details>
<summary>packer log file</summary>

Post the contents of ~/.cache/nvim/packer.nvim.log here

</details>

<details>
<summary>packer compiled file</summary>

Post the contents of `packer_compiled.vim` here
The file is not available, as I deleted it while trying to solve the problem. I deleted all of NeoVim's folders to start fresh thinking that it would solve the problem. But it did not.
</details>

**Comment by AmeerTaweel** [comment]
I reinstalled packer again, and now it gives this error instead:

```
E5113: Error while calling lua chunk: vim/shared.lua:199: attempt to call field 'format' (a nil value)
```

**Comment by AmeerTaweel** [comment]
If I turn off the line that calls packer.startup(), neovim opens with no errors...

**Comment by nanotee** [comment]
Could a plugin/your config be overriding the `string` global somehow? I wasn't able to reproduce this and I don't see anything in your plugin specification that could trigger that error

**Comment by wbthomason** [comment]
I did recently make some tweaks to the log module, but nothing that should've caused this (I think). I also cannot reproduce this issue.

If you run `:lua print(vim.inspect(string.format))`, what do you see?

**Comment by AmeerTaweel** [comment]
As soon as I saw @nanotee comment I knew what the issue was. I did in fact override the `string` global, by:

```lua
_G.string = {}
```

In one of the lines, so I basically completely removed all of the string methods lol.

Now that I fixed that I get no problems whatsoever. And Packer runs like a charm.

To be honest I feel so stupid right now, but I'm new to Lua, and did not think that I could override language standard library with my code, but now I know :sweat_smile:
